diesel/src/r2d2.rs

```diff
@@ -97,22 +97,24 @@ where
     }
 }
 
-impl<T> SimpleConnection for PooledConnection<ConnectionManager<T>>
+impl<M> SimpleConnection for PooledConnection<M>
 where
-    T: Connection + Send + 'static,
+    M: ManageConnection,
+    M::Connection: Connection + Send + 'static,
 {
     fn batch_execute(&self, query: &str) -> QueryResult<()> {
         (&**self).batch_execute(query)
     }
 }
 
-impl<C> Connection for PooledConnection<ConnectionManager<C>>
+impl<M> Connection for PooledConnection<M>
 where
-    C: Connection<TransactionManager = AnsiTransactionManager> + Send + 'static,
-    C::Backend: UsesAnsiSavepointSyntax,
+    M: ManageConnection,
+    M::Connection: Connection<TransactionManager = AnsiTransactionManager> + Send + 'static,
```

Comment:
Is there any reason why we need `TransactionManager = AnsiTransactionManager` here? That will prevent connection implementation that use another TransactionManager from using this impl. (Yes I know that the original version had the same restrictions.…)

Reply:
Yes. If we remove it we end up with:

```
C: Connection,
C::TransactionManager: TransactionManager<Self>,
```

but the definition of `TransactionManger<C>` requires `C: Connection`, so we've got an infinite loop. https://github.com/rust-lang/rfcs/blob/master/text/1647-allow-self-in-where-clauses.md specifies behavior in this case, but that portion of the RFC has not been implemented